thread.c: declare rb_frame_last_func to fix implicit function error.

rb_frame_last_func() is defined in eval.c but never declared in a
header.  Old C allowed implicit declarations; modern clang (-Werror
-Wimplicit-function-declaration in C99+ mode) rejects this.

--- a/thread.c
+++ b/thread.c
@@ -63,6 +63,8 @@
 #include "ruby/io.h"
 #include "ruby/thread.h"

+ID rb_frame_last_func(void);
+
 #ifndef USE_NATIVE_THREAD_PRIORITY
 #define USE_NATIVE_THREAD_PRIORITY 0
 #define RUBY_THREAD_PRIORITY_MAX 3
